roles/edpm_nova/tasks/configure.yml

```diff
@@ -118,3 +118,40 @@
     mode: '0600'
     owner: nova
     group: nova
+
+- name: Check if compute_id exists in state_path
+  tags:
+    - configure
+    - nova
+  become: true
+  ansible.builtin.stat:
+    path: /var/lib/nova/compute_id
+  register: compute_id
+
+- name: Pre-generate compute_id if not already exists
+  tags:
+    - configure
+    - nova
+  become: true
+  ansible.builtin.copy:
+    dest: "/var/lib/nova/compute_id"
+    owner: nova
+    group: nova
+    mode: "0400"
+    attributes: "+i"
+    content: |
+      {{ lookup('ansible.builtin.pipe', 'uuidgen') }}
+  when: not compute_id.stat.exists
```

Comment:
@gibi so one thing to keep in mind is this will break adoption with upsream nova, and downstream until we complete the backprot of the 17.1 change to create the file.

that should be in 17.1.2 i think or 17.1.3

for now with adoption we can simply add a step to create this file with the correct compute service id but i don't know what to do about rdo.

we either need to backport this in upstream nova or our downstream backport need to be put into rdo too

Reply:
/o\ you are right it will break adoption

Reply:
I think the way forward is to add an adoption step as from ansible we cannot get the existing UUID.

Reply:
Adoption step is merged